File: Tutorial4-8LayerImplementation2/src/main/java/com/test/tutorial4_8layerimplementation2/di/AppModule.java
package com.test.tutorial4_8layerimplementation2.di;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import com.test.tutorial4_8layerlibrary.di.BaseSubComponent;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module(subcomponents = {BaseSubComponent.class})
public class AppModule {

    @Singleton
    @Provides
    SharedPreferences provideSharedPreferences(Application application) {
        return application.getSharedPreferences("PrefName", Context.MODE_PRIVATE);
    }

}

File: Tutorial6-4ComponentFactoryDagger-Android/src/main/java/com/smarttoolfactory/tutorial6_4componentfactorydagger_android/di/AppModule.kt
package com.smarttoolfactory.tutorial6_4componentfactorydagger_android.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Provides
    @Singleton
    fun provideSharedPreferences(application: Application): SharedPreferences {
        return application.getSharedPreferences("PrefName", Context.MODE_PRIVATE)
    }

}